clippy_lints/src/doc/mod.rs

```diff
@@ -338,6 +339,30 @@ declare_clippy_lint! {
     "suspicious usage of (outer) doc comments"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Detects documentation that is empty.
+    /// ### Why is this bad?
+    /// It is unlikely that there is any reason to have empty documentation for an item
```

Comment:
Actually there may be multiple reasons to have empty documentation:

* The programmer might have been interrupted after starting the comment
* The programmer might have added an empty doc to circumvent the `missing_docs` lint.
* The programmer might have partially deleted the docs, leaving only an empty doc comment by mistake.

So with that said, there is no valid reason to *keep* empty documentation on an item. Perhaps we should rephrase this part.